README.md

```diff
@@ -223,895 +103,270 @@ const config = [
+registry.defineInjector('router', new Injector(router));
 ```
 
-### Appending routes
-
-With the `router` from the previous example:
+Finally the `registry` needs to be made available to all widgets within the application by setting it as a `property` to the applications top level `Projector` instance.
 
 ```ts
-import Route from '@dojo/routing/Route';
-
-router.append(new Route({ path: '/' }));
-router.append(new Route({ path: '/about' }));
+const projector = new Projector();
+projector.setProperties({ registry });
 ```
 
-These routes won't (yet) do anything.
+#### History Managers
 
-You can append multiple routes at once:
+Routing comes with three history managers for monitoring and changing the navigation state, `HashHistory`, `StateHistory` and `MemoryHistory`. By default the `HashHistory` is used, however this can be overridden by passing a different `HistoryManager` when creating the `Router`.
 
 ```ts
-router.append([
-	new Route({ path: '/' }),
-	new Route({ path: '/about' })
-]);
+const router = new Router(config, MemoryHistory);
 ```
 
-Routes can only be appended to a router once.
+##### Hash History
 
-### Dispatching paths
-
-The router doesn't track navigation events by itself. Changed paths need to be dispatched by application code. Context must be provided, this is made available to the matched routes.
+The hash-based manager uses the fragment identifier to store navigation state and is the default manager used within `@dojo/routing`.
 
 ```ts
-import { Context } from '@dojo/routing/interfaces';
+import { Router } from '@dojo/routing/Router';
+import { HashHistory } from '@dojo/routing/history/HashHistory';
 
-interface AppContext extends Context {
-	someKey: string;
-}
 
-const context: AppContext = {
-	someKey: 'someValue'
-};
-
-router.dispatch(context, '/about');
+const router = new Router(config, HashHistory);
 ```
 
-Route selection starts in a future turn. An async Task is returned (see [`@dojo/core`](https://github.com/dojo/core)) which is resolved with a result object. The object has a `success` property which is `false` if no route was selected, or dispatch was canceled. It's `true` otherwise.
+The history manager has `current` getter, `set(path: string)` and `prefix(path: string)` APIs. The `HashHistory` class assumes the global object is a browser `window` object, but an explicit object can be provided. The manager uses `window.location.hash` and adds an event listener for the `hashchange` event. The `current` getter returns the current path, without a # prefix.
 
-An optional `redirect` property may be present, in case one of the matched routes requested a redirect. The value of the `redirect` property is the new path. It may be an empty string. No routes are executed when a redirect is returned, instead you're expected to change the path and call `dispatch()` again.
+##### State History
 
-You can cancel the task in case a new navigation event occurs.
+State History is coming soon!
```

Comment:
The API is coming soon, or the documentation is coming soon?

Reply:
Oh I have added this, but forgot to add to the readme